atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -37,17 +37,21 @@ public MetricsManager() {
         this.registeredMetrics = new HashSet<>();
     }
 
-    public void registerMetric(Class clazz, String metricPrefix, String metricName, Gauge gauge) {
-        registerMetricWithFqn(MetricRegistry.name(clazz, metricPrefix, metricName), gauge);
+    public MetricRegistry getRegistry() {
+        return metricRegistry;
     }
 
-    public void registerMetric(Class clazz, String metricName, Gauge gauge) {
-        registerMetricWithFqn(MetricRegistry.name(clazz, metricName), gauge);
+    public void registerMetric(Class clazz, String metricPrefix, String metricName, Metric metric) {
```

Comment:
Instead of having to explicitly cast everywhere, you could have a `registerGauge(Class, String, Gauge)` method alongside `registerMetric(Class, String, Metric)`.